cve_bin_tool/version_scanner.py

```diff
@@ -129,12 +131,16 @@ def scan_file(self, filename):
         # parse binary file's strings
         if inpath("strings"):
             # use "strings" on system if available (for performance)
-            o = subprocess.check_output(["strings", filename])
-            lines = o.decode("utf-8").splitlines()
+            s = subprocess.check_output(["strings", filename])
+            lines = s.decode("utf-8").splitlines()
         else:
             # Otherwise, use python implementation
             s = Strings(filename)
             lines = s.parse()
+        #  If python package then strip the lines to avoid detecting other product strings
+        if "PKG-INFO: " in o or "METADATA: " in o:
+            lines = lines[1:3]
+            lines[0] += lines[1]
```

Comment:
The strings here are concatenated to make it unique so a sample checker code would be

```
from cve_bin_tool.checkers import Checker


class Urllib3Checker(Checker):
    CONTAINS_PATTERNS = []
    FILENAME_PATTERNS = [r"METADATA", r"PKG-INFO"]
    VERSION_PATTERNS = [r"Name: urllib3Version: ([0-9]+\.[0-9]+\.[0-9]+)"]
    VENDOR_PRODUCT = [("python", "urllib3")
```

Reply:
This is clever and honestly will probably mostly work, but I'd rather have some sort of explicit mention of cve-bin-tool than a conveniently missing space to make it more clear what's going on.  Something like:

```
 + "--generated pattern for cve-bin-tool " + "Name: urllib3" + Version: "
```

So you'd wind up with a version pattern that looked like

```
    VERSION_PATTERNS = [r"--generated pattern for cve-bin-tool Name: urllib3 Version: ([0-9]+\.[0-9]+\.[0-9]+)"]
```

you can mess with the wording or make the whole thing look more pretty but ideally it'll mention cve-bin-tool and indicate somehow that we made this string appear (and people can refer to the docs for more on how and why).